samcli/lib/intrinsic_resolver/intrinsic_property_resolver.py

```diff
@@ -265,9 +265,14 @@ def resolve_resources_assets(self, ignore_errors):
                             asset.source_path, ignore_errors, parent_function=asset.source_property
                         )
                     elif isinstance(asset, ImageAsset):
-                        asset.source_local_image = self.intrinsic_property_resolver(
-                            asset.source_local_image, ignore_errors, parent_function=asset.source_property
-                        )
+                        if asset.source_local_image:
+                            asset.source_local_image = self.intrinsic_property_resolver(
+                                asset.source_local_image, ignore_errors, parent_function=asset.source_property
+                            )
+                        else:
+                            asset.source_path = self.intrinsic_property_resolver(
+                                asset.source_path, ignore_errors, parent_function=asset.source_property
```

Comment:
@mildaniel Just one thing I want you to double check - will `asset.source_path` be `None` by any chance? (I recall `source_path` is populated whenever `ImageAsset` is instantiated, so just to double confirm).
Or we can be more defensive - we check if `asset.source_path` is also `None`. If it's also `None`, that means we don't have any intrinsic property to resolve

Reply:
As per our conversation, one of source_path or source_local_image will be not None so this should be safe.